src/ompl/geometric/planners/bitstar/src/BITstar.cpp

```diff
@@ -914,11 +909,32 @@ namespace ompl
             //A new start pointer
             const ompl::base::State* newStart;
 
+            //If the list of goal vertices is empty, wait until the first goal state is available.
+            if (goalVertices_.empty())
+            {
+                newGoal = Planner::pis_.nextGoal(ptc);
+                if (newGoal)
+                {
+                    //It is valid, allocate the vertex pointer
+                    goalVertices_.push_back(boost::make_shared<Vertex>(Planner::si_, opt_));
+
+                    //Copy the value into the state
+                    Planner::si_->copyState(goalVertices_.back()->state(), newGoal);
+
+                    //And add this goal to the set of samples:
+                    this->addSample(goalVertices_.back());
+
+                    //Mark that we've added:
+                    addedState = true;
+                }
+            }
+
             //Add the new starts and goals to the lists of said vertices.
             //Do goals first, as they are only added as samples.
-            //If newGoal is NULL, then no new goal exists. The following line both assigns newGoal and checks if it is a valid pointer.
-            while (static_cast<bool>((newGoal = Planner::pis_.nextGoal(ptc))) == true)
+            while (pis_.haveMoreGoalStates())
```

Comment:
What do you mean by "if there isn't another goal"? 
In fact, nextGoal returns NULL only if the ptc becomes true or if the MoveIt! goal sampling thread is inactive. However, even if the maximum number of goal states has been found, this thread remains always active (and do nothing). So the while condition becomes false only when the allowed planning time is passed.